pkg/resources/custom_diffs.go

```diff
@@ -270,3 +270,15 @@ func RecreateWhenStreamIsStale() schema.CustomizeDiffFunc {
 		return nil
 	}
 }
+
+// TODO: [SNOW-1763442] unable to test now, as there is no test accounts with different regions
+// RecreateWhenSecondaryConnectionChangedExternally detects if the secondary connection was promoted externally to serve as primary.
+// If so, it sets the `is_primary` field to `false` which is our desired value for secondary_connection
+func RecreateWhenSecondaryConnectionPromotedExternally() schema.CustomizeDiffFunc {
+	return func(_ context.Context, diff *schema.ResourceDiff, _ any) error {
+		if _, newValue := diff.GetChange("is_primary"); newValue.(bool) {
+			return diff.SetNew("is_primary", false)
```

Comment:
hmmm, this logic is not aligned with others (check RecreateWhenUserTypeChangedExternally, RecreateWhenSecretTypeChangedExternally, or RecreateWhenResourceTypeChangedExternally) - there is ForceNew missing - it's a problem but it can be handled in the next PR too